R/QC_images.R

```diff
@@ -164,20 +164,27 @@ QC_images = function(sample_meta, raw_counts, annotated_counts, normalized_count
   run_below= F
   if(run_below) {
     # Determine which seq cols are present.
-    # sample_meta sequencing columns
-    sm_seq_cols= c('flowcell_name', 'flowcell_lane', 'IndexBarcode1', 'IndexBarcode2')
-    # raw_counts sequencing columns
-    rc_seq_cols= c('flowcell_name', 'flowcell_lane', 'index_1', 'index_2')
+    if("index_2" %in% colnames(sample_meta)){
+      # sample_meta sequencing columns
+      sm_seq_cols= c('flowcell_name', 'flowcell_lane', 'index_1', 'index_2')
+      # raw_counts sequencing columns
+      rc_seq_cols= c('flowcell_name', 'flowcell_lane', 'index_1', 'index_2')
+    }else{
+      sm_seq_cols= c('flowcell_name', 'flowcell_lane', 'index_1')
+      # raw_counts sequencing columns
+      rc_seq_cols= c('flowcell_name', 'flowcell_lane', 'index_1')
+    }
+    
     
     mapped_reads= annotated_counts %>% 
       dplyr::distinct(pick(any_of(c(sm_seq_cols, 'forward_read_cl_barcode', 'expected')))) %>%
       dplyr::mutate(mapped=T)
     
     meta_joining_vector= list()
     for(item in seq_cols) {
-      if(item=='IndexBarcode1') {
+      if(item=='index_1') {
```

Comment:
how is it that filter raw reads never threw an error till now after we had updated the colnames to index1?

Reply:
it is used in the forever false wrapped part of QC images